crates/proto/src/serialize/binary/decoder.rs

```diff
@@ -62,7 +65,7 @@ impl<'a> BinDecoder<'a> {
     /// assert_eq!(decoder.len(), 1);
     /// ```
     pub fn len(&self) -> usize {
-        self.buffer.len().saturating_sub(self.index)
+        self.buffer.len().saturating_sub(self.index())
```

Comment:
The indirection with `self.index()` seems confusing rather than helpful. Isn't `len()` just `self.remaining.len()`?

Reply:
Yes :facepalm: